### PR TITLE
Add base-install docker image

### LINKED ISSUE #2803: Create a base Docker image for our microservices
In #2796 we're starting to use `pnpm` in our microservice Dockerfiles.  Instead of repeating this same change to a dozen Dockerfiles, let's create a base Telescope Service Docker image that includes everything we need for any of our Microservices.

I've done a bunch of updates to how a Dockerfile should look in https://github.com/Seneca-CDOT/telescope/blob/master/src/api/sso/Dockerfile, which we could simplify and adapt.

CC'ing @raygervais in case he has advice.

### PR DESCRIPTION
**Pull request by Kevan-Y**
## Issue This PR Addresses
Fixes #2803 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
A draft version of how to have a base file in docker

Need feedback if I'm doing the docker thing correctly.

**Note:** CI will fail.

## Steps to test the PR

   - `cd src/api/`
   - `docker build --tag telescope-service-base  . ` (Create a image call telescope-service-base)
   - Change a services like image ` src\api\image\Dockerfile`
```DockerFile
    FROM telescope-service-base:latest as deploy
    
    WORKDIR /ome/node/app
    
    COPY --chown=node:node . /home/node/app/
    
    RUN pnpm install --prod
    
    CMD ["node", "src/server.js"]
```
   - ` cd ../../` (back to root workspace)
   - `pnpm services:start image` (It should start image services and use the local image called telescope-service-base)
    
## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/api/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM node:lts-alpine
+
+RUN npm i -g pnpm
```
[the remark this answers is not in the document]
Comment (Kevan-Y) [reply src/api/Dockerfile:4]: Moved into the base
```yml
RUN apk --no-cache add tini
USER node
```

File: src/api/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM node:lts-alpine
+
+# Install pnpm
+RUN npm i -g pnpm
+
+# https://github.com/krallin/tini
+RUN apk --no-cache add tini
```
Comment (humphd) [inline src/api/Dockerfile:7]: This should be:

```dockerfile
# Tini Entrypoint for Alpine
# util-linux required to optimize builds using multiple cores
RUN apk add --no-cache tini util-linux
ENTRYPOINT [ "/sbin/tini", "--"]
```

By setting an `ENTRYPOINT` you are [defining the default command that gets run in the container](https://docs.docker.com/engine/reference/builder/#entrypoint).  In this case, we're saying, "use tini and have it receive its arguments from what comes next (--)", see https://unix.stackexchange.com/questions/11376/what-does-double-dash-mean.  The `tini` command will take care of initializing our node process for us.

A Dockerfile that uses this as a base would then define a `CMD` to [specify default params to our ENTRYPOINT](https://docs.docker.com/engine/reference/builder/#cmd):

```dockerfile
CMD ["node", "index.js"]
```

So in essence, we're really doing:

```sh
/sbin/tini node index.js
```

But we split it into stages, and the second stage can redefine what the actually arguments are via `CMD`.

**Review by humphd — CHANGES_REQUESTED** [review]

**Comment by Kevan-Y** [comment]
@humphd I think I will revert back the change of DockerFile for the image. I will only push the DockerFile for the base. I will make a follow-up issue for changing the rest of the services later when the base image is hosted into a registry

**Review by humphd — CHANGES_REQUESTED** [review]
Yes, get rid of the changes to the `image/Dockerfile` until we have this in the registry.  The rest looks good.

**Review by DukeManh — APPROVED** [review]

**Review by humphd — APPROVED** [review]

**Pull request merged**